src/compute/src/lib.rs

```diff
@@ -107,6 +70,46 @@ pub struct ComputeNodeOpts {
     /// The parallelism that the compute node will register to the scheduler of the meta service.
     #[clap(long, default_value_t = default_parallelism())]
     pub parallelism: usize,
+
+    #[clap(flatten)]
+    override_config: OverrideConfigOpts,
```

Comment:
Yes. The intention of a separate field is to stress that "the fields in this struct should not be used after `load_config`", since `load_config` will move this field.

Reply:
I can fully understand we don't want to let all CLI opts be a sys config, but why do we restrict that only some config is overrideable? 🤔

Reply:
IMO the main reason is not all items in the config file apply to one worker type. We're currently using a shared toml schema to simplify the deployment, so the above situation is unavoidable on the file part. But if you type `compactor_node --help` and see `barrier_interval`, it would feel weird, as CLI args has a strong indication of "this argument must take effect".

That being said, if we use a different schema for each worker type, I think allowing overriding everything is ok. But maybe we can keep things simple for now 🤔

Reply:
makes sense

Reply:
Do you want to take another look at the implementation, since I'm not using layered configuration lib? If not I will merge this PR.

Reply:
Didn't take a detailed look but I'm OK if it works. And I begin to understand your points. ✅